lib/plugins/blocks.js

```diff
@@ -254,9 +238,15 @@ function inject (bot, { version, storageBuilder }) {
     if (!block) return null
 
     if (extraInfos) {
-      block.signText = signs[block.position]
       block.painting = paintingsByPos[block.position]
-      block.blockEntity = blockEntities[block.position]
+      if (block.entity) {
+        const id = block.entity.value.id.value
+        // Mege in abstractions related to signs
+        Object.assign(block.blockEntity, addExtraBlockEntityData(id, block.entity))
+        if (id === 'minecraft:sign' || id === 'Sign') {
```

Comment:
this should be in pchunk/pworld

this is basically a more high level way to represent sign and there could be something like that for other block entities in the future